### PR TITLE
gRPC echo and Websocket (DVWS) apps have been added

### PR DESCRIPTION
**Pull request by 416e64726579**
There are two new web applications "echo-server" and "DVWS" which are based on gRPC and websocket correspondingly
https://github.com/grpc/grpc-web/
https://github.com/interference-security/DVWS
Usage an envoy version of Wallarm along with sidecar container
https://hub.docker.com/r/wallarm/envoy
https://github.com/416e64726579/docker-wallarm-node/tree/feature/sidecar
https://hub.docker.com/r/awallarm/wallarm-node-sidecar/tags

### CODE REVIEW

File: terraform/common/variables.tf [hunk 1]
```diff
@@ -116,3 +116,13 @@ variable "wordpress_password" {
   type        = string
   description = "Wordpress admin username"
 }
+
+variable "deploy_user" {
```
Comment (vgartvich-wallarm) [inline terraform/common/variables.tf]: "waf_node_deploy_username"
"waf_node_deploy_password"

File: terraform/demo-apps/dvws/default [hunk 2]
```diff
@@ -0,0 +1,61 @@
+map $remote_addr $wallarm_mode_real {
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: map -> geo

File: terraform/demo-apps/dvws/default [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  # Please replace <WALLARM_MODE> below by the request filtering mode:
+  # off to disable request processing
+  # monitoring to process but not block requests
+  # block to process all requests and block the malicious ones
+  default monitoring;
+  # IP addresses and rules for US cloud scanners
+  23.239.18.250 off;104.237.155.105 off;45.56.71.221 off;45.79.194.128 off;104.237.151.202 off;45.33.15.249 off;45.33.43.225 off;45.79.10.15 off;45.33.79.18 off;45.79.75.59 off;23.239.30.236 off;50.116.11.251 off;45.56.123.144 off;45.79.143.18 off;172.104.21.210 off;74.207.237.202 off;45.79.186.159 off;45.79.216.187 off;45.33.16.32 off;96.126.127.23 off;172.104.208.113 off;192.81.135.28 off;35.236.51.79 off;35.236.75.97 off;35.236.111.124 off;35.236.108.88 off;35.236.16.246 off;35.236.61.185 off;35.236.110.91 off;35.236.14.198 off;35.235.124.137 off;35.236.48.47 off;35.236.100.176 off;35.236.18.117 off;35.235.112.188 off;35.236.55.214 off;35.236.126.84 off;35.236.3.158 off;35.236.127.211 off;35.236.118.146 off;35.236.20.89 off;35.236.1.4 off;
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: Please take the list from https://github.com/vgartvich-wallarm/gke-demo-app-cluster/blob/master/terraform/system-apps/wallarm-ingress/whitelist.yaml

File: terraform/demo-apps/dvws/default [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+}
+server {
+    listen 56245 default_server;
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: Please add a comment about the server

File: terraform/demo-apps/dvws/default [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    location /healthz {
+            allow 10.0.0.0/8;
+            allow 172.16.0.0/12;
+            allow 192.168.0.0/16;
+            deny all;
+            return 200;
+    }
+}
+server {
+    listen 56246 default_server;
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: Please add a comment about the server

File: terraform/demo-apps/dvws/default [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    listen [::]:56246 default_server ipv6only=on;
+    server_name localhost;
+    root /usr/share/nginx/html;
+    index index.html index.htm;
+    wallarm_mode $wallarm_mode_real;
+    # wallarm_instance 1;
+    set_real_ip_from 0.0.0.0/0;
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: Please set to "set_real_ip_from 10.0.0.0/0;"

File: terraform/demo-apps/dvws/default [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    location / {
+            # Please replace <APP_CONTAINER_PORT> below by the port number
+            # on which the container accepts incoming requests,
+            # the value must be identical to ports.containerPort
+            # in definition of your main app container
+            proxy_pass http://localhost:80;
+            include proxy_params;
+    }
+    location /healthz {
+            allow 10.0.0.0/8;
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: Please remove the four allow/deny lines.

File: terraform/demo-apps/dvws/default [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    }
+}
+server {
+    listen 56246 default_server;
+    listen [::]:56246 default_server ipv6only=on;
+    server_name localhost;
+    root /usr/share/nginx/html;
+    index index.html index.htm;
+    wallarm_mode $wallarm_mode_real;
+    # wallarm_instance 1;
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: Please use app instance 3

File: terraform/demo-apps/dvws/default [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    set_real_ip_from 0.0.0.0/0;
+    real_ip_header X-Forwarded-For;
+    location / {
+            # Please replace <APP_CONTAINER_PORT> below by the port number
+            # on which the container accepts incoming requests,
+            # the value must be identical to ports.containerPort
+            # in definition of your main app container
+            wallarm_mode monitoring;
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/default]: Please change to the block mode.

File: terraform/demo-apps/dvws/main.tf [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+            items {
+              key  = "default"
+              path = "default"
+            }
+          }
+        }
+
+        container {
+          name    = "wallarm-nginx"
+          image   = "awallarm/wallarm-node-sidecar:slim"
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/main.tf]: Please follow https://docs.wallarm.com/admin-en/installation-guides/kubernetes/wallarm-sidecar-container-manifest/

File: terraform/demo-apps/dvws/main.tf [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+          }
+
+          port {
+            name           = "ws"
+            container_port = 56246
+          }
+
+          env {
+            name  = "WALLARM_API_HOST"
+            value = "api.wallarm.com"
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/main.tf]: Pls use var.api_host (?)

File: terraform/demo-apps/dvws/main.tf [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+            }
+
+            initial_delay_seconds = 3
+            period_seconds        = 3
+          }
+
+          image_pull_policy = "Always"
+        }
+
+        container {
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/main.tf:119]: Please remove the Tarantool container.

File: terraform/demo-apps/dvws/main.tf [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+
+            value_from {
+              secret_key_ref {
+                name = "tf-wallarm-node-secret"
+                key  = "WALLARM_API_TOKEN"
+              }
+            }
+          }
+
+          resources {
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/main.tf]: Please remove all resource limits.

File: terraform/demo-apps/dvws/main.tf [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+            name           = "http"
+            container_port = 80
+          }
+
+          port {
+            name           = "ws"
+            container_port = 8080
+          }
+
+          resources {
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/dvws/main.tf]: Pls remove resource limits.

File: terraform/demo-apps/grpc-web/build/common/Dockerfile [hunk 15]
```diff
@@ -0,0 +1,21 @@
+FROM node:10-stretch
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/grpc-web/build/common/Dockerfile]: Move all Dockerfile to vgartvich-wallarm/grpc-demo repo.

File: terraform/demo-apps/grpc-web/build/envoy/envoy.yaml [hunk 16]
```diff
@@ -0,0 +1,140 @@
+admin:
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/grpc-web/build/envoy/envoy.yaml]: Please move the file to demo-app/grpc-web/ directory.

File: terraform/demo-apps/grpc-web/main.tf [hunk 17]
```diff
@@ -0,0 +1,222 @@
...
+      name        = "front-http"
+      port        = 80
+      target_port = 8081
+    }
+    port {
+      name        = "back-http"
+      port        = 8080
+      target_port = 8080
+    }
+    port {
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/grpc-web/main.tf]: Pls remove the 9090 service.

File: terraform/demo-apps/grpc-web/build/envoy/envoy.yaml [hunk 18]
```diff
@@ -0,0 +1,140 @@
...
+            name: local_route
+            virtual_hosts:
+            - name: backend
+              ########################################
+              # wallarm filter config per virtual host
+              ########################################
+              typed_per_filter_config:
+                wallarm:
+                  "@type": type.googleapis.com/wallarm.WallarmConf
+                  instance: 42 # mark everything within this VHost as instance 42
```
Comment (vgartvich-wallarm) [inline terraform/demo-apps/grpc-web/build/envoy/envoy.yaml]: Pls user app instance 2

**Review by vgartvich-wallarm — CHANGES_REQUESTED** [review]

**Comment by 416e64726579** [comment]
I have fixed everything according to the comments. The new variables were to be added for an ordinary Wallarm Node container.

**Pull request merged**